axlearn/common/checkpointer.py

```diff
@@ -272,6 +298,7 @@ def __init__(self, cfg: Config):
             self._manager = array_serialization.GlobalAsyncCheckpointManager(
                 timeout_secs=cfg.timeout_secs
             )
+        self._executor = futures.ThreadPoolExecutor()
```

Comment:
Since we're no longer using a context manager, should we invoke `shutdown` explicitly?

Reply:
Technically we don't because when executor is garbage collected, it will signal its worker threads to stop.

Reply:
We generally should avoid relying on implicitly cleanup at the mercy of the garbage collector. Let's add the `shutdown` when the checkpointer exits the context manager.

Reply:
A `stop` method makes sense to me. Thanks!

Reply:
Added.